### PR TITLE
PLG-243: Enable Appcues onboarding for the FT

### PR DESCRIPTION
**Pull request by olivierpontier**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/FreeTrial/back/Infrastructure/Symfony/Resources/config/requirejs.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+config:
+  paths:
+    pim/onboarding: akeneofreetrial/js/onboarding/appcuses-onboarding.ts
```
Comment (AymericPlanche) [inline src/Akeneo/FreeTrial/back/Infrastructure/Symfony/Resources/config/requirejs.yml:3]:
```suggestion
    pim/onboarding: akeneofreetrial/js/onboarding/appcues-onboarding.ts
```

File: src/Akeneo/FreeTrial/back/Infrastructure/Symfony/Resources/public/js/onboarding/appcuses-onboarding.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+const AppcuesOnBoarding: PimOnBoarding = {
+  registerUser: () => {
+    getAppcuesAgent().then(appcues => {
+      if (!FeatureFlags.isEnabled('free_trial') || appcues === null) {
+        return;
+      }
+
+      appcues.identify(UserContext.get('meta').id, {
+        username: UserContext.get('username'),
+        email: UserContext.get('email'),
```
Comment (AymericPlanche) [inline src/Akeneo/FreeTrial/back/Infrastructure/Symfony/Resources/public/js/onboarding/appcuses-onboarding.ts:17]: do not send email to appcues to ease RGPD compliancy. 
the only unique user identifier must be the PIM username.
you can pass other pim user related data but every "personnal" data must be challenged.

File: src/Akeneo/FreeTrial/back/Infrastructure/Symfony/Resources/public/js/onboarding/appcuses-onboarding.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import {getAppcuesAgent} from './appcues-agent';
+import {PimOnBoarding} from 'pimui/js/onboarding';
+
+const FeatureFlags = require('pim/feature-flags');
+const UserContext = require('pim/user-context');
+const Mediator = require('oro/mediator');
+
+const AppcuesOnBoarding: PimOnBoarding = {
```
Comment (AymericPlanche) [inline src/Akeneo/FreeTrial/back/Infrastructure/Symfony/Resources/public/js/onboarding/appcuses-onboarding.ts:8]:
```suggestion
const AppcuesOnboarding: PimOnboarding = {
```

no need to camelcase, it's a word ! ;)

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/onboarding/pim-onboarding.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import {PimOnBoarding} from './index';
+
+const DefaultPimOnBoarding: PimOnBoarding = {
```
Comment (AymericPlanche) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/onboarding/pim-onboarding.ts:3]: no camelcase (everywhere)

File: src/Akeneo/FreeTrial/back/Infrastructure/Symfony/Resources/public/js/onboarding/appcuses-onboarding.ts (hunk 2 above)
Comment (AymericPlanche) [reply src/Akeneo/FreeTrial/back/Infrastructure/Symfony/Resources/public/js/onboarding/appcuses-onboarding.ts:17]: we can discuss about it tomorrow ;)

**Review by AymericPlanche — CHANGES_REQUESTED** [review]

**Review by AymericPlanche — APPROVED** [review]

**Review by LaurentPetard — APPROVED** [review]

**Pull request merged**